internal/contour/virtualhost.go

```diff
@@ -190,16 +189,36 @@ func action(i *v1beta1.Ingress, be *v1beta1.IngressBackend) *route.Route_Route {
 
 // actioncrd computes the cluster route action, a *v2.Route_route for the
 // supplied ingress and backend
-func actioncrd(namespace string, be *v1alpha1.Upstream) *route.Route_Route {
+func actioncrd(namespace string, be []v1alpha1.Upstream) *route.Route_Route {
 
-	name := ingressBackendToClusterName(namespace, be.ServiceName, strconv.Itoa(be.ServicePort))
+	upstreams := []*route.WeightedCluster_ClusterWeight{}
+
+	// Loop over all the upstreams and add to slice
+	for _, i := range be {
+
+		name := ingressBackendToClusterName(namespace, i.ServiceName, strconv.Itoa(i.ServicePort))
+
+		upstream := route.WeightedCluster_ClusterWeight{
+			Name: name,
+			Weight: &google_protobuf1.UInt32Value{
+				Value: uint32(math.Floor(100 / float64(len(be)))),
+			},
+		}
+
+		upstreams = append(upstreams, &upstream)
+	}
+
+	// Create Route with slice of upstreams
 	ca := route.Route_Route{
 		Route: &route.RouteAction{
-			ClusterSpecifier: &route.RouteAction_Cluster{
-				Cluster: name,
+			ClusterSpecifier: &route.RouteAction_WeightedClusters{
```

Comment:
I vaguely remember there being a `TotalWeight` field somewhere in this type. Do we need to set that?

Reply:
The doc says, 
```
// Specifies the total weight across all clusters. The sum of all cluster weights must equal this
// value, which must be greater than 0. Defaults to 100.
```
I _suspect_ that it exists to make the weight math easier. Ie if you have three backends with 33% each, then you set the total weight to 99, not 100.

I'd say leave it for this PR, but add a TODO to keep a running tally of the weights set in each weightedcluster and assign it to TotalWeight as a sanity check.

Reply:
added a todo